### PR TITLE
My loop revamp

### PR DESCRIPTION
**Pull request by winterstein**
This isn't ready yet - but making a PR anyway, as I think its the best way to pass comments back and forth.

### CODE REVIEW

File: src/style/AccountPage.less [hunk 1]
```diff
@@ -15,12 +15,101 @@
...
+
+	.user {
+		img {
+			width: 100%;
+			@media @md-max {
+				width: 25%;
+				margin-bottom: 1rem;
+			}
+		}
+		h1 {
```
Comment (winterstein) [inline src/style/AccountPage.less:28]: Redefining fonts for a specific section is usually a mistake, as it leads to inconsistent styling + harder to maintain css.

Can we remove this?
Should it be promoted to a universal site-wide rule?

If this is needed as a specific patch, please leave a brief comment for why.

File: src/style/AccountPage.less [hunk 2]
```diff
@@ -15,12 +15,101 @@
...
+			font-size: 1.6rem;
+			font-weight: 600;
+		}
+	}
+
+	h1, h2, h3, h4 {
```
Comment (winterstein) [inline src/style/AccountPage.less:34]: As above -- Can we remove this? If not, why? Thanks

NB: this comment applies for every font edit really, but I won't repeat it as that'd be tiresome.

File: src/style/AccountPage.less [hunk 3]
```diff
@@ -15,12 +15,101 @@
...
+	h2,h4 {
+		font-weight: 600;
+		text-transform: uppercase;
+	}
+
+	h2 {
+		color: @gl-light-red;
+	}
+
+	p {
```
Comment (winterstein) [inline src/style/AccountPage.less:50]: Section specific overrides for p and a are also to be avoided by default. As above, remove, or promote, or comment. Thanks.

File: src/style/AccountPage.less [hunk 4]
```diff
@@ -15,12 +15,101 @@
...
+					width: 50% !important;
+				}
+			}
+			.doline {
+				position: absolute;
+				top: 50%;
+				bottom: 0;
+				left: 8.3%;
+				padding: 0px 1.5px;
+				background: #F9B139;
```
Comment (winterstein) [inline src/style/AccountPage.less:74]: Please use variables for colours. Thanks.

File: src/style/AccountPage.less [hunk 5]
```diff
@@ -15,12 +15,101 @@
...
+			.do-img {
+				@media @md-max {
+					width: 50% !important;
+				}
+			}
+			.doline {
+				position: absolute;
+				top: 50%;
+				bottom: 0;
+				left: 8.3%;
```
Comment (winterstein) [inline src/style/AccountPage.less:72]: 50% is self explanatory, but 8.3% should have a comment.

File: src/style/AccountPage.less [hunk 6]
```diff
@@ -15,12 +15,101 @@
...
+	.settings {
+		font-size: 1.5rem;
+		a {
+            color: #126CFC !important;
+            text-decoration: underline !important;
+		}
+		small {
+			font-size: 0.8rem;
+		}
+		.hover-info {
```
Comment (winterstein) [inline src/style/AccountPage.less:96]: This css is probably creating some kind of widget. In which case, it maybe should be refactored out of the specific AccountPage settings context.

File: src/style/AccountPage.less [hunk 7]
```diff
@@ -15,12 +15,101 @@
...
+	}
+
+	.settings {
+		font-size: 1.5rem;
+		a {
+            color: #126CFC !important;
+            text-decoration: underline !important;
+		}
+		small {
+			font-size: 0.8rem;
```
Comment (winterstein) [inline src/style/AccountPage.less:94]: remove / promote / comment

File: src/style/CampaignPage.less [hunk 8]
```diff
@@ -204,6 +211,26 @@ body {
 		font-size: 0.8rem;
 	}
 
+	.theunit {
+		z-index: 1;
+		width: 95%;
+		top: 50%;
+		left:50%;
+		transform: translate(-50%, -50%);
```
Comment (winterstein) [inline src/style/CampaignPage.less:219]: This is doing something clever -- What?

File: src/style/GetInvolvedPage.less [hunk 9]
```diff
@@ -0,0 +1,73 @@
+.GetInvolvedPage {
+    .mission p {
+        a {
+            text-decoration: none !important;
+            transition: 0.3s;
+            transform: scale(1) rotate(0deg);
+            display: inline-block;
+            &:hover {
+                transform: scale(2) rotate(20deg);
```
Comment (winterstein) [inline src/style/GetInvolvedPage.less:9]: Comment please

File: src/style/GetInvolvedPage.less [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+                transition: 0.3s;
+            }
+        }
+    }
+    .w-md-50 {
```
Comment (winterstein) [inline src/style/GetInvolvedPage.less:14]: These look like handy utility classes. Promote them to their own top-level file?

File: src/style/LandingSection.less [hunk 11]
```diff
@@ -6,34 +6,47 @@
 	width: 100%;
 	display: flex;
 
+	@media @md-max {
+		.share-btn {
+			bottom: 100px !important;
+			left: 50% !important;
+			transform: translateX(-50%);
+			z-index: 999;
+		}
```
Comment (winterstein) [inline src/style/LandingSection.less:15]: Comment - What is this solving?  Thanks

File: src/style/navbar.less [hunk 12]
```diff
@@ -24,14 +41,25 @@
...
-		.login-link {
-			color: #AD2016;
-			& a {
-				color: #AD2016;
-			}
-		}
+	}
+
+	.nav-link:not(.btn) {
+		text-transform: uppercase;
```
Comment (winterstein) [inline src/style/navbar.less:50]: How does this look in the Impact Hub pages?

File: src/style/buttons.less [hunk 13]
```diff
@@ -1,135 +1,131 @@
...
+    color: @gl-red !important;
+    box-shadow: none !important;
+    border: solid 2px @gl-red !important;
+}
+
+/*
+.btn-transparent-white {
+    background-color: transparent;
+    color: white !important;
+    padding: 13px 38px; // Account for border
```
Comment (winterstein) [inline src/style/buttons.less:109]: Sizing in a colour class seems odd.

File: web/index.html [hunk 14]
```diff
@@ -19,6 +19,19 @@
...
+			(function(h,u,b){
+			var d=h.getElementsByTagName("script")[0],e=h.createElement("script");
+			e.async=true;e.src='https://cookiehub.net/c2/1ac697d7.js';
+			e.onload=function(){u.cookiehub.load(b);}
+			d.parentNode.insertBefore(e,d);
+			})(document,window,cpm);
+			</script>
+		
+		<!-- Detect ad-blockers -->
+		<script src="/doubleclick.js"></script>
```
Comment (winterstein) [inline web/index.html:34]: See DetectAdBlock

File: src/js/components/ShareButton.jsx [hunk 15]
```diff
@@ -11,48 +11,42 @@ import { encURI } from '../base/utils/miscutils';
  * description
  */
 class ShareButton extends React.Component {
```
Comment (winterstein) [inline src/js/components/ShareButton.jsx:13]: This looks good -- I love the popout tray of icons.

Why does it use `class extends React.Component`?   
It should work fine as a plain jsx function, with useState() for tray open/closed.

**Comment by dodo721** [comment]
Going to resolve the comments and refactoring issues - but releasing first for prioritisation

**Pull request merged**